src/demo.rs

```diff
@@ -0,0 +1,111 @@
+use notify::{Config, Event, RecommendedWatcher, RecursiveMode, Watcher};
+use std::path::Path;
+use std::path::PathBuf;
+use std::thread;
+use tokio::fs::metadata;
+use tokio::fs::File;
+use tokio::io::{AsyncReadExt, AsyncSeekExt, SeekFrom};
+use tokio::sync::mpsc;
+use tokio::time::interval;
+use tokio::time::Duration;
+use serde::{Serialize, Deserialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BigBrotherPacket {
+    players: Vec<String>, // List of players on the server, array of SteamIDs
+    ip: String, // IP address of the server
+    sequence: i32, // A sequence number that starts at 0 and increments for each successive fragment
+    demo: Option<Vec<u8>>, // Up-to-date copy of demo file. None for packet 0 as the demo wouldn't be flushed to disk yet.
+}
+
+
+pub async fn demo_loop(demo_path: PathBuf) -> anyhow::Result<()> {
+    let (sync_tx, sync_rx) = std::sync::mpsc::channel::<Event>();
+    let (tx, mut rx) = mpsc::channel(32);
+    let config = Config::default().with_poll_interval(Duration::from_secs(2));
+
+    let mut watcher: RecommendedWatcher = Watcher::new(
+        Box::new(move |res: Result<Event, notify::Error>| {
+            let _ = sync_tx.send(res.unwrap());
```

Comment:
Should probably `tracing::error!` and then ignore any `Err`s that arrive instead of unwrapping.